### ISSUE TITLE
Email as attachment

### ISSUE THREAD

**Issue by hugoroy** [astroidmail/astroid#237]
Having trouble dealing with email as attachments.

Let's say I want to forward 2 emails from 2 different threads. This is a bit difficult to do with any email app.

1. Go to email 1, save email 1 in a file with `s` (astroid saves email1.eml)
2. Go to email 2, save email 2 in a file with `s` (astroid saves email2.eml)
3. Write a new email, attach email1.eml and email2.eml

Now, things go wrong.

**Scenario 1**: Send the email right away

Now if you try to open your sent email and open an attachment, you get this:

![capture d ecran de 2016-12-03 13-10-15](https://cloud.githubusercontent.com/assets/1629771/20859265/e6f4e3cc-b959-11e6-9b2a-0b09b6b34565.png)

**Scenario 2**: Save as draft

Now go back to your draft, edit the email body in your text editor, save the email body text. And the attachments are gone.

**Comment by gauteh** [comment]
Hm, maybe the emails have to be attached as GMimeMessage parts. Hopefully this can be done in a reliable way, will check.

**Comment by gauteh** [comment]
Please try #239.

**Comment by hugoroy** [comment]
It worked. Apart from the fact that when saving as draft, it was then written:

" Error: Non-viewable part! " in the body of the email

**Comment by gauteh** [comment]
Hugo Roy writes on desember 7, 2016 21:27:
> It worked. Apart from the fact that when saving as draft, it was then written:
> 
> " Error: Non-viewable part! " in the body of the email

What did you do to get that?

**Comment by hugoroy** [comment]
1. save an email as file.eml
2. write a new email
3. attach file.eml
4. save as draft
5. open draft again
6. edit the draft
7. here it is.

**Comment by hugoroy** [comment]
1. save an email as file.eml
2. write a new email
3. attach file.eml
4. save as draft
4. quit astroid
5. open draft again
6. edit the draft
7. here it is.

**Comment by gauteh** [comment]
Hm, it doesn't here - would you mind trying with attaching another e-mail?

**Comment by hugoroy** [comment]
↪ Gaute Hope, décembre 7, 2016 9:37 :
> Hm, it doesn't here - would you mind trying with attaching another e-mail?

OK, this only happens when saving/attaching an email which also has an
HTML part (like a github-issue email notification).

**Comment by gauteh** [comment]
Is the re-start of astroid necessary?

**Comment by gauteh** [comment]
Hugo Roy writes on desember 7, 2016 21:56:
> ↪ Gaute Hope, décembre 7, 2016 9:37 :
>> Hm, it doesn't here - would you mind trying with attaching another e-mail?
> 
> OK, this only happens when saving/attaching an email which also has an
> HTML part (like a github-issue email notification).

I am unable to re-produce this.. trying with these github issues with
html parts. I've updated the branch a little, could you try pull --force
and see if it is still an issue?

**Comment by hugoroy** [comment]
To reproduce the bug, yes.



Le 7 décembre 2016 22:29:01 GMT+01:00, Gaute Hope <notifications@github.com> a écrit :
>Is the re-start of astroid necessary?

-- 
Envoyé de mon appareil mobile. 
Sent from my mobile device.

**Comment by gauteh** [comment]
Hugo Roy writes on desember 8, 2016 0:55:
> To reproduce the bug, yes.

Does this happen if you do not have a signature attached? Do you have
the signature inline or attached?

**Comment by gauteh** [comment]
I do not understand how this is possible. It should be enough to save
the draft and completely close the message before re-opening it. There
isn't kept any cache anywhere of the message.

Hugo Roy writes on desember 8, 2016 0:55:
> To reproduce the bug, yes.
>

**Comment by gauteh** [comment]
So I think I can reliably re-produce this. And I believe the bug happens
somewhere in `ComposeMessage::load_message`. It is not necessary to
re-start astroid:

1. save a multipart e-mail with a HTML part (like github comment)
2. create new email
3. attach multipart email
4. save draft
5. close draft
6. open draft
7. edit
8. when saving after the part is messed up
9. trying to save the draft causes trouble, in my case gmime write gets
   in to an infinite loop. writing 'Content-type: message/rfc822'...
   again and again.

**Comment by gauteh** [comment]
Please try #239 again.

**Comment by hugoroy** [comment]
Opening an email thread failed:

```



[10:47:10.437175] [0x00007f5c973c3a00] [debug] mw: focus-in: 1 active: true, in_quit: false
[10:47:10.890998] [0x00007f5c973c3a00] [debug] ti: open thread: 0000000000023845 (true)
[10:47:10.893417] [0x00007f5c973c3a00] [debug] theme: loading..
[10:47:10.893556] [0x00007f5c973c3a00] [info] re: using user configured resource: /home/hugo/.gitRepos/astroid/ui/thread-view.html
[10:47:10.893728] [0x00007f5c973c3a00] [debug] tv: testing version: 3
[10:47:10.893803] [0x00007f5c973c3a00] [error] tv: html file version does not match!
[10:47:10.893896] [0x00007f5c973c3a00] [info] re: using user configured resource: /home/hugo/.gitRepos/astroid/ui/thread-view.scss
[10:47:10.894005] [0x00007f5c973c3a00] [debug] tv: testing version: 3
[10:47:10.894060] [0x00007f5c973c3a00] [error] tv: scss file version does not match!
[10:47:10.894920] [0x00007f5c973c3a00] [info] theme: processing: /home/hugo/.config/astroid/ui/thread-view.scss
[10:47:10.895554] [0x00007f5c973c3a00] [error] theme: error processing: Internal Error: File context has no input path


(astroid:26308): glibmm-ERROR **: 
unhandled exception (type std::exception) in signal handler:
what: theme: could not process scss: Internal Error: File context has no input path


[1]    26308 trace trap (core dumped)  ./astroid
```

**Comment by gauteh** [comment]
This is the libsass bug, try rebase off master.


fre. 30. des. 2016 kl. 10.48 skrev Hugo Roy <notifications@github.com>:

**Comment by gauteh** [comment]
I've updated the branch, pull-force and you should be good.

**Comment by gauteh** [comment]
Merged #239, considering fixed.
